test/src/DeviceRequestPrompt.spec.ts

```diff
@@ -0,0 +1,457 @@
+      const client = new MockCDPSession();
+      const timeoutSettings = new TimeoutSettings();
+      const manager = new DeviceRequestPromptManager(client, timeoutSettings);
+
+      await expect(
+        manager.waitForDevicePrompt({timeout: 1})
+      ).rejects.toBeInstanceOf(TimeoutError);
+    });
+
+    it('should respect default timeout when there is no custom timeout', async () => {
+      const client = new MockCDPSession();
+      const timeoutSettings = new TimeoutSettings();
+      const manager = new DeviceRequestPromptManager(client, timeoutSettings);
+
+      timeoutSettings.setDefaultTimeout(1);
+      await expect(manager.waitForDevicePrompt()).rejects.toBeInstanceOf(
+        TimeoutError
+      );
+    });
+
+    it('should prioritize exact timeout over default timeout', async () => {
+      const client = new MockCDPSession();
+      const timeoutSettings = new TimeoutSettings();
+      const manager = new DeviceRequestPromptManager(client, timeoutSettings);
+
+      timeoutSettings.setDefaultTimeout(0);
+      await expect(
+        manager.waitForDevicePrompt({timeout: 1})
+      ).rejects.toBeInstanceOf(TimeoutError);
+    });
+
+    it('should work with no timeout', async () => {
+      const client = new MockCDPSession();
+      const timeoutSettings = new TimeoutSettings();
+      const manager = new DeviceRequestPromptManager(client, timeoutSettings);
+
+      const [prompt] = await Promise.all([
+        manager.waitForDevicePrompt({timeout: 0}),
+        (async () => {
+          await new Promise(resolve => {
+            setTimeout(resolve, 50);
+          });
+          client.emit('DeviceAccess.deviceRequestPrompted', {
+            id: '00000000000000000000000000000000',
+            devices: [],
+          });
+        })(),
+      ]);
+      expect(prompt).toBeTruthy();
+    });
+
+    it('should return the same prompt when there are many watchdogs simultaneously', async () => {
+      const client = new MockCDPSession();
+      const timeoutSettings = new TimeoutSettings();
+      const manager = new DeviceRequestPromptManager(client, timeoutSettings);
+
+      const [prompt1, prompt2] = await Promise.all([
+        manager.waitForDevicePrompt(),
+        manager.waitForDevicePrompt(),
+        (() => {
+          client.emit('DeviceAccess.deviceRequestPrompted', {
+            id: '00000000000000000000000000000000',
+            devices: [],
+          });
+        })(),
+      ]);
+      expect(prompt1 === prompt2).toBeTruthy();
+    });
+
+    it('should listen and shortcut when there are no watchdogs', async () => {
+      const client = new MockCDPSession();
+      const timeoutSettings = new TimeoutSettings();
+      const manager = new DeviceRequestPromptManager(client, timeoutSettings);
+
+      client.emit('DeviceAccess.deviceRequestPrompted', {
+        id: '00000000000000000000000000000000',
+        devices: [],
+      });
+
+      expect(manager).toBeTruthy();
```

Comment:
@mzgoddard, sorry for quite a late comment. But what's the purpose of this `expect`?
How the `manager` won't be truthy?